client_code/messaging.py

```diff
@@ -20,46 +25,49 @@ def __init__(self, subscriber, handler):
 
 
 class Publisher:
-    def __init__(self, with_logging=True):
-        self.with_logging = with_logging
+    def __init__(self, logger: _Logger = None, **kwargs):
+        if logger is None:
+            logger = _Logger(name="publisher", format="{msg}")
+        self.logger = logger
```

Comment:
```python

class Publisher:
    default_logger = Logger(name="publisher", format="{msg}")
    def __init__(self, logger: _Logger = None, **kwargs):
        self.logger = logger or self.default_logger
```

This way we can also change the log level of the default logger
like 

```python
from anvil_extras.messaging import Publisher
from anvil_extras.logging import ERROR

Publisher.default_logger.log_level = ERROR
# or
Publisher.default_logger = Logger(...)
```

But I think this api suffers from not having an easy way to turn off logging.

Reply:
Yeah, I agree, but every solution that I think goes back to having the `with_logging` parameter or having an hybrid approach like the first one that I did.

Also, I changed the docs to show info about the usage of loggers in the messaging module.

Will fix what you commented right now.

Reply:
yeah curious to hear @meatballs thoughts on this api

Reply:
For disabling the log, what about returing the `with_logging` option to each method, so that you can disable for specific calls. 

We could also add `with_logging` to the constructor and set logger to None if disabled. 

Additionally, we could create a `classmethod` to disable the `default_logger`, setting it to None so that you could disable it (or change it) globally, like in other modules.

Then, before each logging action, we check if `self.logger` is None (or with_logging is True).

Reply:
Yeah. And actually most of the time you probably want logging off by default. 

If you could add updating the docs to this PR that'd be great.

Reply:
The default logger could be a null logger. i.e. if you pass no logger, you get no logging.